lib/template-parts.php

```diff
@@ -191,15 +182,52 @@ function filter_rest_wp_template_part_collection_params( $query_params ) {
  */
 function filter_rest_wp_template_part_query( $args, $request ) {
 	if ( $request['theme'] ) {
-		$meta_query   = isset( $args['meta_query'] ) ? $args['meta_query'] : array();
-		$meta_query[] = array(
-			'key'   => 'theme',
-			'value' => $request['theme'],
+		$tax_query   = isset( $args['tax_query'] ) ? $args['tax_query'] : array();
+		$tax_query[] = array(
+			'taxonomy' => 'wp_theme',
+			'field'    => 'slug',
+			'terms'    => $request['theme'],
 		);
 
-		$args['meta_query'] = $meta_query;
+		$args['tax_query'] = $tax_query;
 	}
 
 	return $args;
 }
 add_filter( 'rest_wp_template_part_query', 'filter_rest_wp_template_part_query', 99, 2 );
+
+/**
+ * Filters the post data for a response.
+ *
+ * @param WP_REST_Response $response The response object.
+ * @return WP_REST_Response
+ */
+function filter_rest_prepare_wp_template_part( $response ) {
+	if ( isset( $response->data ) && is_array( $response->data ) && isset( $response->data['id'] ) ) {
+		$response->data['wp_theme_slug'] = false;
+
+		// Get the wp_theme terms.
+		$wp_themes = wp_get_post_terms( $response->data['id'], 'wp_theme' );
+
+		// If a theme is assigned, add it to the REST response.
+		if ( $wp_themes && is_array( $wp_themes ) ) {
+			$wp_theme_slugs = array_column( $wp_themes, 'slug' );
+
+			$file_based                   = in_array( 'wp_file_based', $wp_theme_slugs, true );
+			$response->data['file_based'] = $file_based;
+
+			$theme_slug = array_filter(
+				$wp_theme_slugs,
+				function( $slug ) {
+					return 'wp_file_based' !== $slug;
+				}
+			);
+			if ( $theme_slug ) {
+				$response->data['wp_theme_slug'] = $theme_slug[0];
+			}
+		}
+	}
+
+	return $response;
+}
+add_filter( 'rest_prepare_wp_template_part', 'filter_rest_prepare_wp_template_part' );
```

Comment:
Do we still need this?
The `wp_theme` taxonomy is set to `show_in_rest` so we should get it for free now.

Reply:
Easier to access this way. I'd like to avoid adding JS code to every part where we accessed the meta. We would need to `select` the taxonomy terms and do filtering and all kind of stuff.

Reply:
Ah right, the term ID fooled me twice. 😤